test/standalone.test.ts

```diff
@@ -0,0 +1,144 @@
+          },
+          completion: {
+            completionItem: {
+              snippetSupport: true,
+            },
+          },
+          semanticTokens: {
+            tokenTypes: [],
+            tokenModifiers: [],
+            formats: [],
+            requests: {},
+          },
+          formatting: {
+            dynamicRegistration: true,
+          },
+        },
+      },
+      workspaceFolders: null,
+    }
+    const logger = new NullLogger()
+    if (!serverFork.stdin || !serverFork.stdout) {
+      throw 'Process created without stdio streams'
+    }
+    clientConnection = createProtocolConnection(
+      new StreamMessageReader(serverFork.stdout as Duplex),
+      new StreamMessageWriter(serverFork.stdin as Duplex),
+      logger
+    )
+    clientConnection.onError((err) => {
+      console.error('ERROR', err)
+    })
+    clientConnection.onNotification(LogMessageNotification.type, (params) => {
+      if (params.type !== 3) {
+        logMessages.push(params)
+      }
+    })
+    clientConnection.onUnhandledNotification((n) => {
+      console.error('Unhandled notification', n)
+    })
+    clientConnection.listen()
+    const { serverInfo } = await clientConnection.sendRequest(
+      InitializeRequest.type,
+      initializeParams
+    )
+    assert.strictEqual(serverInfo?.name, 'Cucumber Language Server')
+  })
+
+  afterEach(() => {
+    clientConnection.end()
+    clientConnection.dispose()
+    serverFork.kill('SIGTERM') // Try to terminate first
+    serverFork.kill('SIGKILL') // Then try to kill if it is not killed
+  })
+
+  context('workspace/didChangeConfiguration', () => {
+    it(`startup success`, async () => {
+      // First we need to configure the server, telling it where to find Gherkin documents and Glue code.
+      // Note that *pushing* settings from the client to the server is deprecated in the LSP. We're only using it
+      // here because it's easier to implement in the test.
+      const settings: Settings = {
+        features: ['testdata/**/*.feature'],
+        glue: ['testdata/**/*.js'],
+        parameterTypes: [],
+        snippetTemplates: {},
+      }
+      const configParams: DidChangeConfigurationParams = {
+        settings,
+      }
+
+      await clientConnection.sendNotification(DidChangeConfigurationNotification.type, configParams)
+
+      await new Promise((resolve) => setTimeout(resolve, 1000))
+
+      assert.strictEqual(
+        logMessages.length,
+        // TODO: change this to 0 when `workspace/semanticTokens/refresh` issue was solved
+        1,
+        // Pretty log messages
+        logMessages
+          .map(({ type, message }) => `**Type**: ${type}\n**Message**:\n${message}\n`)
+          .join('\n--------------------\n')
+      )
+    })
+  })
+})
+
+class NullLogger implements Logger {
+  error(): void {
+    // no-op
+  }
+
+  warn(): void {
+    // no-op
+  }
+
+  info(): void {
+    // no-op
+  }
+
+  log(): void {
+    // no-op
+  }
+}
```

Comment:
As this code is shared with `CucumberLanguageServer.test.ts`, would it be possible to extract the function into a separate module and import to both?

https://github.com/cucumber/language-server/blob/c2fbe5f66112d2e25df252094ecd8182c7bacf7b/test/CucumberLanguageServer.test.ts#L210-L226